### PR TITLE
[Artist] Only include displayable shows.

### LINKED ISSUE #473: [PartnerShow] Nested under an artist, may include non-displayable shows
Re https://github.com/artsy/eigen/issues/1873

While a single partner show [will be verified to be displayable](https://github.com/artsy/metaphysics/blob/6bbb98f4640861c6fa0e61a84dd40e0872218136/schema/partner_show.js#L273), shows nested under an artist [do not](https://github.com/artsy/metaphysics/blob/6bbb98f4640861c6fa0e61a84dd40e0872218136/schema/artist/index.js#L464-L468). This leads to cases where shows are returned that e.g. don’t have a cover image, which triggers the following error:

```
TypeError: Cannot read property 'id' of null
    at resolve (schema/partner_show.js:330:61)
```

### PR DESCRIPTION
**Pull request by alloy**
Depends on https://github.com/artsy/gravity/pull/10566
Fixes #473 & https://github.com/artsy/eigen/issues/1873

### CODE REVIEW

**Comment by mzikherman** [comment]
I have a feeling this is now going to break/undo the work to make reference shows visible on the Artist page.

For instance: https://github.com/artsy/force/blob/master/apps/artist/queries/cv.coffee

We do want to be able to return those reference shows for the CV listing. 

It seems like with the filter you added in https://github.com/artsy/gravity/pull/10566, `displayable: false` will return _all_ shows.

So is the correct thing to make Metaphysics allow that to be specified by a client (but can default to true), and then we have to update Force/Microgravity to pass in false where it wants to be able to fetch/display non-displayable shows?

Basically as-is, I _think_ this will no longer allow those reference shows to show up on the front-end.

**Comment by alloy** [comment]
I see what you mean, good catch :+1:

**Comment by alloy** [comment]
> It seems like with the filter you added in https://github.com/artsy/gravity/pull/10566, `displayable: false` will return all shows.

No it has to be `nil`, so undefined, for all shows to show up.

**Comment by mzikherman** [comment]
> No it has to be nil, so undefined, for all shows to show up.

Ah right, `nil` in the ES query, which you get by leaving the query param out entirely. `?displayable=true` is only displayable shows, and `?displayable=false` is only non displayable shows. Got it.

**Comment by alloy** [comment]
Ok, so this is embarrassing, I don’t actually believe that `displayable` is the real solution here. In retrospect the issue is that there’s no partner information for these shows. I’ve updated the PR to only guard against that (and have removed the use of the `displayable` param).

I don’t actually think that guarding here is the real proper fix, because a lot of code in `PartnerShow` will assume a partner and it seems kind of obvious that there should be such data. Could it be that this is a case that you mentioned where the partner might have been removed from Galaxy?

**Comment by mzikherman** [comment]
Oy, now I'm confused.

You can have shows that have a `galaxy_partner_id`, and won't have any `partner_id` or `partner` in the JSON.

The new [Show](https://github.com/artsy/metaphysics/blob/6bbb98f4640861c6fa0e61a84dd40e0872218136/schema/show.js#L124-L146) schema allows for that, but there's lots of copy-pasta from `PartnerShow`, so I think we should update that to be more defensive in all the places a partner is assumed, and generally switch to using that.

So the issue is that shows w/o partners (and only a `galaxy_partner_id` set) are coming back? And causing crashes in querying other fields that do `partner.id`? In that case, since those partner-less shows _are_ something we support and are valid upstream in Gravity/API-land, I think the right fix (which I should have done/realized but totally didn't!), is to make things more defensive to allow for that, as you've done here.

There are _probably_ a few other places in the `PartnerShow` and `Show` schemas where a similar defense should technically exist.
